test/broadway_test.exs

```diff
@@ -582,6 +589,61 @@ defmodule BroadwayTest do
 
       refute_received {:EXIT, _, ^processor}
     end
+
+    test "telemetry", c do
+      self = self()
+
+      :ok =
+        :telemetry.attach_many(
+          "#{c.test}",
+          [
+            [:broadway, :processor, :start],
+            [:broadway, :processor, :stop],
+            [:broadway, :consumer],
+            [:broadway, :batcher]
+          ],
+          fn name, measurements, metadata, _ ->
+            send(self, {:got, name, measurements, metadata})
+          end,
+          nil
+        )
+
+      ref = Broadway.test_messages(c.broadway, [1, 2, :fail, :fail_batcher], batch_mode: :bulk)
+      assert_receive {:batch_handled, [%{data: 1, status: :ok}, %{data: 2, status: :ok}]}
+      assert_receive {:batch_handled, [%{data: :fail_batcher, status: :ok}]}
+
+      assert_receive {:ack, ^ref, [%{status: :ok}, %{status: :ok}], []}
+      assert_receive {:ack, ^ref, [], [%{status: {:failed, "Failed message"}}]}
+      assert_receive {:ack, ^ref, [], [%{status: {:failed, "Failed batcher"}}]}
+
+      assert_receive {:got, [:broadway, :processor, :start], %{}, %{}}
+      assert_receive {:got, [:broadway, :processor, :stop], %{}, metadata}
+      assert [] = metadata.failed_messages
+
+      assert_receive {:got, [:broadway, :processor, :start], %{}, %{}}
+      assert_receive {:got, [:broadway, :processor, :stop], %{}, %{failed_messages: []}}
+      assert [] = metadata.failed_messages
+
+      assert_receive {:got, [:broadway, :processor, :start], %{}, %{}}
+      assert_receive {:got, [:broadway, :processor, :stop], %{}, metadata}
+      assert [%{data: :fail}] = metadata.failed_messages
+
+      assert_receive {:got, [:broadway, :processor, :start], %{}, %{}}
+      assert_receive {:got, [:broadway, :processor, :stop], %{}, metadata}
+      assert [] = metadata.failed_messages
+
+      assert_receive {:got, [:broadway, :consumer], %{}, metadata}
+      assert [] = metadata.failed_messages
+
+      assert_receive {:got, [:broadway, :consumer], %{}, metadata}
+      assert [%{data: :fail_batcher}] = metadata.failed_messages
+
+      assert_receive {:got, [:broadway, :batcher], %{}, %{events: [_]}}
+      assert_receive {:got, [:broadway, :batcher], %{}, %{events: [_]}}
+      assert_receive {:got, [:broadway, :batcher], %{}, %{events: [_]}}
+
+      refute_receive _
```

Comment:
We should avoid having this since it slows down tests. I'd go with just not having anything here.